ospfd/ospf_lsa.c

```diff
@@ -2214,7 +2214,7 @@ struct ospf_lsa *ospf_external_lsa_originate(struct ospf *ospf,
 
 	/* Create new AS-external-LSA instance. */
 	if ((new = ospf_external_lsa_new(ospf, ei, NULL)) == NULL) {
-		if (IS_DEBUG_OSPF_EVENT)
+		if (ei && IS_DEBUG_OSPF_EVENT)
```

Comment:
Above we can have the same situation, right? 

```c
	if (ospf->router_id.s_addr == INADDR_ANY) {
		if (IS_DEBUG_OSPF_EVENT)
			zlog_debug(
				"LSA[Type5:%pI4]: deferring AS-external-LSA origination, router ID is zero",
				&ei->p.prefix);
		return NULL;
	}
```

Reply:
yes

Reply:
Should we fix this too?